sway/input/cursor.c

```diff
@@ -0,0 +1,184 @@
+#define _XOPEN_SOURCE 700
+#ifdef __linux__
+#include <linux/input-event-codes.h>
+#elif __FreeBSD__
+#include <dev/evdev/input-event-codes.h>
+#endif
+#include <wlr/types/wlr_cursor.h>
+#include <wlr/types/wlr_xcursor_manager.h>
+#include "sway/input/cursor.h"
+#include "sway/view.h"
+#include "list.h"
+#include "log.h"
+
+static void cursor_update_position(struct sway_cursor *cursor) {
+	double x = cursor->cursor->x;
+	double y = cursor->cursor->y;
+
+	wlr_xcursor_manager_set_cursor_image(cursor->xcursor_manager,
+		"left_ptr", cursor->cursor);
```

Comment:
There's a bug in wlroots about setting a hardware cursor's image twice (it fallbacks to software cursors, I'll fix that).

Setting the cursor image when it isn't needed is inefficient because it involves uploading the image to the GPU each time. This needs to be fixed here (maybe keep the name of the currently selected cursor in the struct?).

Reply:
I dunno I'm not going to do the cursor until last and all of this will probably change. I'm happy if it just shows up.

Reply:
Okay just add a TODO then